### PR TITLE
Compare graph warning when only one graph is opened

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

This feature adds a notification to the user when only one graph is present and an attempt is made to compare graphs.
This below image shows the notification that pops up when the compare graph is clicked and one graph exists only. This will not happen if two graphs are present. This will also not be possible to open two graphs, then compare graph, and close one graph. The compare graph is modal and therefore this shouldn't happen.

image

### Alternate Designs
I looked to add custom logic to the context handler, but this is not accessible within our code it seems. 
This seemed like a good way to go about it because the user should have no need to compare with one graph open. This will descriptively let them know that to compare you need more than one graph.

### Why Should This Be In Core?
Usability

### Benefits
Usability

### Possible Drawbacks
None realised

### Verification Process
Opened no graphs, one graph, and many. 
Tried closing the graph once compare graph was opened. Modal so could not be done.

### Applicable Issues
#1647

### CODE REVIEW

File: CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java [hunk 1]
```diff
@@ -41,8 +45,15 @@ public CompareGraphAction(final GraphNode context) {
 
     @Override
     public void actionPerformed(final ActionEvent e) {
-        PluginExecution.withPlugin(CorePluginRegistry.COMPARE_GRAPH)
+        // Check current graphs opened
+        if(GraphManager.getDefault().getAllGraphs().size() > 1){
```
Comment (antares1470) [inline CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java:49]: Are you able to add a space between `if` and `(`. Similar for `)` and `{`

File: CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java [hunk 2]
```diff
@@ -41,8 +45,15 @@ public CompareGraphAction(final GraphNode context) {
...
+            PluginExecution.withPlugin(CorePluginRegistry.COMPARE_GRAPH)
                 .interactively(true)
                 .executeLater(context.getGraph());
+        } else {
+            Platform.runLater(()-> {
```
Comment (antares1470) [inline CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java:54]: similar for `()` and `->`

**Review by antares1470 — CHANGES_REQUESTED** [review]
Minor requests but looks good otherwise

**Review by Delphinus8821 — APPROVED** [review]

**Review by Auriga2 — CHANGES_REQUESTED** [review]
You can remove the useless curly braces as triggered in code smells. 
Also not clear what you mean by `Tried closing the graph once compare graph was opened. Modal so could not be done.` . I can close one graph and compare again with the new compare graph opened. Should that also populate the same message?

File: CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java (hunk 1 above)
Comment (aldebaran30701) [reply CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/compare/CompareGraphAction.java]: I'll have to look at my onSave() formatting, usually this is auto formatted out. fixed.

**Comment by aldebaran30701** [comment]
@Auriga2 
So I have two graphs opened then i click on the compare graph tool.
With the compare graph tool still opened, I try to click behind it and close one of those graphs. This was an attempt to get a situation where the compare graph tool was opened and only one graph existed. I couldn't click the close button of any graph while the compare graph tool was opened because of the modality of the compare graph tool. It was 100% always in focus for windows tests.

**Comment by Auriga2** [comment]
Ah got it. I got confused with the graph opened when comparing. The Compare graph tool window remains focused as you mentioned.

**Review by Auriga2 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**